File: src/Z.Core/System.String/String.ToPlural.cs
// Description: C# Extension Methods | Enhance the .NET Framework and .NET Core with over 1000 extension methods.
// Website & Documentation: https://csharp-extension.com/
// Issues: https://github.com/zzzprojects/Z.ExtensionMethods/issues
// License (MIT): https://github.com/zzzprojects/Z.ExtensionMethods/blob/master/LICENSE
// More projects: https://zzzprojects.com/
// Copyright © ZZZ Projects Inc. All rights reserved.
#if !NETSTANDARD
using System.Data.Entity.Design.PluralizationServices;
using System.Globalization;
#endif

public static partial class Extensions
{
#if !NETSTANDARD
    /// <summary>
    ///     A string extension method that converts the @this to a plural.
    /// </summary>
    /// <param name="this">The @this to act on.</param>
    /// <returns>@this as a string.</returns>
    public static string ToPlural(this string @this)
    {
        return PluralizationService.CreateService(new CultureInfo("en-US")).Pluralize(@this);
    }

    /// <summary>
    ///     A string extension method that converts the @this to a plural.
    /// </summary>
    /// <param name="this">The @this to act on.</param>
    /// <param name="cultureInfo">Information describing the culture.</param>
    /// <returns>@this as a string.</returns>
    public static string ToPlural(this string @this, CultureInfo cultureInfo)
    {
        return PluralizationService.CreateService(cultureInfo).Pluralize(@this);
    }
#endif
}

File: src (VB.NET)/Z.ExtensionMethods.VB/Z.Core/System.String/String.ToPlural.vb

' Copyright (c) 2015 ZZZ Projects. All rights reserved
' Licensed under MIT License (MIT) (https://github.com/zzzprojects/Z.ExtensionMethods)
' Website: http://www.zzzprojects.com/
' Feedback / Feature Requests / Issues : http://zzzprojects.uservoice.com/forums/283927
' All ZZZ Projects products: Entity Framework Extensions / Bulk Operations / Extension Methods /Icon Library

Imports System.Data.Entity.Design.PluralizationServices
Imports System.Globalization

Public Module Extensions_536

	''' <summary>
	'''     A string extension method that converts the @this to a plural.
	''' </summary>
	''' <param name="this">The @this to act on.</param>
	''' <returns>@this as a string.</returns>
	<System.Runtime.CompilerServices.Extension> _
	Public Function ToPlural(this As String) As String
		Return PluralizationService.CreateService(New CultureInfo("en-US")).Pluralize(this)
	End Function

	''' <summary>
	'''     A string extension method that converts the @this to a plural.
	''' </summary>
	''' <param name="this">The @this to act on.</param>
	''' <param name="cultureInfo">Information describing the culture.</param>
	''' <returns>@this as a string.</returns>
	<System.Runtime.CompilerServices.Extension> _
	Public Function ToPlural(this As String, cultureInfo As CultureInfo) As String
		Return PluralizationService.CreateService(cultureInfo).Pluralize(this)
	End Function
End Module


